base/reflection.jl

```diff
@@ -1084,17 +1084,38 @@ function parentmodule(@nospecialize(f), @nospecialize(types))
 end
 
 """
-    hasmethod(f, Tuple type; world = typemax(UInt)) -> Bool
+    hasmethod(f, t::Type{<:Tuple}[, kwnames]; world=typemax(UInt)) -> Bool
 
 Determine whether the given generic function has a method matching the given
 `Tuple` of argument types with the upper bound of world age given by `world`.
 
+If a tuple of keyword argument names `kwnames` is provided, this also checks
+whether the method of `f` matching `t` has the given keyword argument names.
+If the matching method accepts a variable number of keyword arguments, e.g.
+with `kwargs...`, the method's keyword arguments must be a subset of the
```

Comment:
This doesn't seem right; `f(; a, b, c...)` supports keyword args `(:a,)` even though {a,b} isn't a subset of {a}. I guess if there is a `...` keyword arg this should unconditionally return `true`.